From e6d6e23baea0e4e3fb969ac054fc197d21a663ba Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 27 Nov 2024 16:41:48 +0800
Subject: [PATCH 1/3] fix(fpm): define perferences for AOSC OS

- Use syslog (-> journald) infrastructure to output server logs.
- Listen /run/php-fpm/php-fpm.sock to integrate with systemd.
- Set default server root at /srv/http, consistent with httpd.
- Enable non-root daemon owners users and groups.
- Set php-fpm PID file path to /run/php-fpm/php-fpm.pid (instead of
  ./run/php-fpm.pid).
- Install configurations at /etc/php-fpm.conf and /etc/php-fpm.d/www.conf.
---
 sapi/fpm/Makefile.frag      | 4 ++--
 sapi/fpm/php-fpm.conf.in    | 4 ++--
 sapi/fpm/php-fpm.service.in | 2 +-
 sapi/fpm/www.conf.in        | 8 ++++----
 4 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/sapi/fpm/Makefile.frag b/sapi/fpm/Makefile.frag
index c6a290f9d59b..b96d7f11079e 100644
--- a/sapi/fpm/Makefile.frag
+++ b/sapi/fpm/Makefile.frag
@@ -15,8 +15,8 @@ install-fpm: $(SAPI_FPM_PATH)
 	else \
 		echo "Installing PHP FPM defconfig:     $(INSTALL_ROOT)$(sysconfdir)/" && \
 		$(mkinstalldirs) $(INSTALL_ROOT)$(sysconfdir)/php-fpm.d; \
-		$(INSTALL_DATA) sapi/fpm/php-fpm.conf $(INSTALL_ROOT)$(sysconfdir)/php-fpm.conf.default; \
-		$(INSTALL_DATA) sapi/fpm/www.conf $(INSTALL_ROOT)$(sysconfdir)/php-fpm.d/www.conf.default; \
+		$(INSTALL_DATA) sapi/fpm/php-fpm.conf $(INSTALL_ROOT)$(sysconfdir)/php-fpm.conf; \
+		$(INSTALL_DATA) sapi/fpm/www.conf $(INSTALL_ROOT)$(sysconfdir)/php-fpm.d/www.conf; \
 	fi
 
 	@echo "Installing PHP FPM man page:      $(INSTALL_ROOT)$(mandir)/man8/"
diff --git a/sapi/fpm/php-fpm.conf.in b/sapi/fpm/php-fpm.conf.in
index f1b48adca08f..c212d728a7e2 100644
--- a/sapi/fpm/php-fpm.conf.in
+++ b/sapi/fpm/php-fpm.conf.in
@@ -14,14 +14,14 @@
 ; Pid file
 ; Note: the default prefix is @EXPANDED_LOCALSTATEDIR@
 ; Default Value: none
-;pid = run/php-fpm.pid
+;pid = /run/php-fpm/php-fpm.pid
 
 ; Error log file
 ; If it's set to "syslog", log is sent to syslogd instead of being written
 ; into a local file.
 ; Note: the default prefix is @EXPANDED_LOCALSTATEDIR@
 ; Default Value: log/php-fpm.log
-;error_log = log/php-fpm.log
+error_log = syslog
 
 ; syslog_facility is used to specify what type of program is logging the
 ; message. This lets syslogd specify that messages from different facilities
diff --git a/sapi/fpm/php-fpm.service.in b/sapi/fpm/php-fpm.service.in
index 50a87dc555f4..64c03bc48765 100644
--- a/sapi/fpm/php-fpm.service.in
+++ b/sapi/fpm/php-fpm.service.in
@@ -8,7 +8,7 @@ After=network.target
 
 [Service]
 Type=@php_fpm_systemd@
-PIDFile=@EXPANDED_LOCALSTATEDIR@/run/php-fpm.pid
+PIDFile=/run/php-fpm/php-fpm.pid
 ExecStart=@EXPANDED_SBINDIR@/php-fpm --nodaemonize --fpm-config @EXPANDED_SYSCONFDIR@/php-fpm.conf
 ExecReload=/bin/kill -USR2 $MAINPID
 
diff --git a/sapi/fpm/www.conf.in b/sapi/fpm/www.conf.in
index 9689e4defab0..d2d23213cdc6 100644
--- a/sapi/fpm/www.conf.in
+++ b/sapi/fpm/www.conf.in
@@ -38,7 +38,7 @@ group = @php_fpm_group@
 ;                            (IPv6 and IPv4-mapped) on a specific port;
 ;   '/path/to/unix/socket' - to listen on a unix socket.
 ; Note: This value is mandatory.
-listen = 127.0.0.1:9000
+listen = /run/php-fpm/php-fpm.sock
 
 ; Set listen(2) backlog.
 ; Default Value: 511 (-1 on Linux, FreeBSD and OpenBSD)
@@ -50,8 +50,8 @@ listen = 127.0.0.1:9000
 ; and group can be specified either by name or by their numeric IDs.
 ; Default Values: Owner is set to the master process running user. If the group
 ;                 is not set, the owner's group is used. Mode is set to 0660.
-;listen.owner = @php_fpm_user@
-;listen.group = @php_fpm_group@
+listen.owner = @php_fpm_user@
+listen.group = @php_fpm_group@
 ;listen.mode = 0660
 
 ; When POSIX Access Control Lists are supported you can set them using
@@ -422,7 +422,7 @@ pm.max_spare_servers = 3
 ; Chdir to this directory at the start.
 ; Note: relative path can be used.
 ; Default Value: current directory or / when chroot
-;chdir = /var/www
+;chdir = /srv/http
 
 ; Redirect worker stdout and stderr into main error log. If not set, stdout and
 ; stderr will be redirected to /dev/null according to FastCGI specs.

base-commit: 9327bec388094ecf44f2bd6db9fe8e6518849b15
-- 
2.48.0.rc1

